src/components/Resource/Forms/Training/TrainingImages.tsx

```diff
@@ -0,0 +1,507 @@
+    // async function parseExisting(ef: (typeof thisModelVersion.files)[number]) {
+    async function parseExisting() {
+      const url = createModelFileDownloadUrl({
+        versionId: thisModelVersion.id,
+        type: 'Training Data',
+      });
+      const result = await fetch(url);
+      if (!result.ok) {
+        return;
+      }
+      const blob = await result.blob();
+      const zipFile = new File([blob], `${thisModelVersion.id}_training_data.zip`, {
+        type: blob.type,
+      });
+      return await handleZip(zipFile, false);
+    }
+    const existingDataFile = thisModelVersion.files.find((f) => f.type === 'Training Data');
+    if (existingDataFile) {
+      setLoadingZip(true);
+      parseExisting().then((files) => {
+        if (files) {
+          const flatFiles = files.flat();
+          setImageList(flatFiles);
+          setInitialImageList(flatFiles.map((d) => ({ ...d })));
+        }
+        setLoadingZip(false);
+      });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const thisStep = 2;
+  const maxImgPerPage = 9;
+
+  const handleZip = async (f: FileWithPath, showNotif = true) => {
+    // could set loadingZip here too
+    const parsedFiles: imageDataType[] = [];
+    const zipReader = new JSZip();
+    // zipReader.loadAsync(f).then((zData) => {
+    const zData = await zipReader.loadAsync(f);
+    await Promise.all(
+      Object.entries(zData.files).map(async ([zname, zf]) => {
+        // - we could read the type here with some crazy blob/hex inspecting
+        const fileSplit = zname.split('.');
+        const fileExt = fileSplit.pop() || '';
+        const baseFileName = fileSplit.join('.');
+        if (fileExt in imageExts) {
+          const imgBlob = await zf.async('blob');
+          const czFile = zipReader.file(`${baseFileName}.txt`);
+          let captionStr = '';
+          if (czFile) captionStr = await czFile.async('string');
+          parsedFiles.push({
+            name: zname,
+            type: imageExts[fileExt],
+            url: URL.createObjectURL(imgBlob),
+            caption: captionStr,
+          });
+        }
+      })
+    );
+
+    if (showNotif) {
+      if (parsedFiles.length > 0) {
+        showSuccessNotification({
+          title: 'Zip parsed successfully!',
+          message: `Found ${parsedFiles.length} images.`,
+          autoClose: 3000,
+        });
+      } else {
+        showErrorNotification({
+          error: new Error('Could not find any valid files in zip.'),
+        });
+      }
+    }
+
+    return parsedFiles;
+  };
+
+  const handleDrop = async (fileList: FileWithPath[]) => {
+    // blob:http://localhost:3000/3949be19-8afa-41ee-9cc6-4773e6a0ab73
+    console.log(fileList);
+
+    const newFiles = await Promise.all(
+      fileList.map(async (f) => {
+        if (
+          ['application/zip', 'application/x-zip-compressed', 'multipart/x-zip'].includes(f.type)
```

Comment:
Perhaps add these to `mime-types.ts`

Reply:
done, but doing this requires me to do: `ZIP_MIME_TYPE.includes(f.type as never)` otherwise I get an error. any way around this?